openhtf/util/units/bin/units_from_xls.py

```diff
@@ -104,31 +204,32 @@ def unit_defs_from_sheet(sheet, column_names):
 
     # loop over all remaining rows and pull out units.
     for row in rows:
-      name = row[col_indices[column_names[0]]].value
+      name = row[col_indices[column_names[0]]].value.replace("'", r'\'')
       code = row[col_indices[column_names[1]]].value
-      suffix = row[col_indices[column_names[2]]].value.replace("'", "\\\'")
-      if name in seen:
+      suffix = row[col_indices[column_names[2]]].value.replace("'", r'\'')
+      key = unit_key_from_name(name, UNIT_KEY_REPLACEMENTS)
+      if key in seen:
         continue
-      seen.add(name)
+      seen.add(key)
 
-      yield "UNITS['%s'] = Unit('%s', '''%s''')\n" % (
-          multi_replace(name.upper(),
-                        UNIT_KEY_REPLACEMENTS),
-                        code,
-                        suffix)
+      yield "%s = UnitInfo('%s', '%s', '''%s''')\n" % (key, name, code, suffix)
+      yield "ALL_UNITS.append(%s)\n" % key
 
   except xlrd.XLRDError:
     sys.stdout.write('Unable to process the .xls file.')
 
 
-def multi_replace(string, replacements=None):
-  """Return the string in uppercase with the specified substitutions."""
+def unit_key_from_name(name, replacements=None):
```

Comment:
still don't need to allow `replacements` to be `None`.

Don't need to copy name using `[:]`, it's immutable. I'd just use `name` throughout to avoid the extra-but-unused variable.